clp_ffi_py/query_builder.py

```diff
@@ -0,0 +1,182 @@
+    is empty.
+
+    For more details about the search query CLP IR stream supports, please read
+    the documents of `clp_ffi_py.ir.Query` and
+    `clp_ffi_py.wildcard_query.WildcardQuery`.
+    """
+
+    def __init__(self) -> None:
+        self._search_time_lower_bound: int = Query.default_search_time_lower_bound()
+        self._search_time_upper_bound: int = Query.default_search_time_upper_bound()
+        self._search_time_termination_margin: int = Query.default_search_time_termination_margin()
+        self._wildcard_queries: List[WildcardQuery] = []
+
+    @property
+    def search_time_lower_bound(self) -> int:
+        return self._search_time_lower_bound
+
+    @property
+    def search_time_upper_bound(self) -> int:
+        return self._search_time_upper_bound
+
+    @property
+    def search_time_termination_margin(self) -> int:
+        return self._search_time_termination_margin
+
+    @property
+    def wildcard_queries(self) -> List[WildcardQuery]:
+        return deepcopy(self._wildcard_queries)
+
+    def set_search_time_lower_bound(self, ts: int) -> QueryBuilder:
+        """
+        Sets the search time lower bound.
+
+        :param ts: The timestamp of search time lower bound.
+        :return: self.
+        """
+        self._search_time_lower_bound = ts
+        return self
+
+    def set_search_time_upper_bound(self, ts: int) -> QueryBuilder:
+        """
+        Sets the search time upper bound.
+
+        :param ts: The timestamp of search time upper bound.
+        :return: self.
+        """
+        self._search_time_upper_bound = ts
+        return self
+
+    def set_search_time_termination_margin(self, ts: int) -> QueryBuilder:
+        """
+        Sets the search time termination margin.
+
+        :param ts: The timestamp of search time termination margin.
+        :return: self.
+        """
+        self._search_time_termination_margin = ts
+        return self
+
+    def add_wildcard_query(self, wildcard_query: str, case_sensitive: bool = False) -> QueryBuilder:
+        """
+        Adds a wildcard query to the builder's wildcard query list.
+
+        :param wildcard_query: The wildcard query string to add.
+        :param case_sensitive: Case sensitive indicator.
+        :return: self.
+        """
+        self._wildcard_queries.append(WildcardQuery(wildcard_query, case_sensitive))
+        return self
+
+    def add_wildcard_queries(self, wildcard_queries: List[WildcardQuery]) -> QueryBuilder:
+        """
+        Adds a list wildcard queries to the builder's wildcard query list.
+
+        :param wildcard_queries: The list of wildcard queries to be added.
+        :return: self.
+        """
+        self._wildcard_queries.extend(wildcard_queries)
+        return self
+
+    def reset_search_time_lower_bound(self) -> QueryBuilder:
+        """
+        Resets the search time lower bound to the default value (the smallest
```

Comment:
Don't mention the default value since it can go out of sync with the code. If you want, you could reference the method (:meth:`~clp_ffi_py.ir.Query.default_search_time_lower_bound`).